src/canvas/canvas.service.ts

```diff
@@ -194,6 +196,24 @@ export class CanvasService {
     });
   }
 
+  public async giveAccessByTag(command: GiveAccessByTagCommand) {
+    const user = await this.userRepository.findOne({
+      where: { id: command.userId },
+    });
+    if (!user) {
+      throw new NotFoundException();
+    }
+    const accessibleWithTags = await this.readByTags(command.tagIds, user.id);
+    accessibleWithTags.forEach((canvas) =>
```

Comment:
Shouldn't we allow only the owner of the canvas to give access? 
Currently everyone that has access can share it to anyone without any limits.

Reply:
@coredumped7893 No. The assumption was that every person with access to the given canvas can give access to it. Example use case: a PM giving access to all canvases tagged with some project name to some developers.

Reply:
That makes sense, but we need to be aware of the fact that anyone with access can give access to anyone else. This will make access control quite painful - how do I make sure that canvas X is visible to only person Y?

Reply:
but for the current scope I think that this logic should be enough 👌🏼